src/Microsoft.Azure.NotificationHubs.DotNetCore.Tests/NotificationHubClientTest.cs

```diff
@@ -739,6 +739,35 @@ private async Task DeleteInstallationAsync_CreateAndDeleteInstallation_Installat
             RecordTestResults();
         }
 
+        [Fact]
+        private async Task InstallationExists_ChecksInstallation()
+        {
+            LoadMockData();
+            await DeleteAllRegistrationsAndInstallations();
+
+            var installationId = Guid.NewGuid().ToString();
+
+            var installation = new Installation
+            {
+                InstallationId = installationId,
+                Platform = NotificationPlatform.Apns,
+                PushChannel = _configuration["AppleDeviceToken"]
+            };
+
+            await _hubClient.CreateOrUpdateInstallationAsync(installation);
+
+            await Task.Delay(TimeSpan.FromSeconds(1));
```

Comment:
I understand why the delay is necessary when running the tests against real endpoints, but it definitely wastes time when running tests against the response snapshots.

Can we create a `async Task Delay(TimeSpan)` helper on the test class and make it no-op when not recording test results? I want to avoid adding seconds to the test run time every time we add a new test case.

Reply:
:+1: we should be able to use the same flag we use for determining whether the run is live or mocked